test/e2e/workload_test.go

```diff
@@ -0,0 +1,301 @@
+	})
+	var (
+		shouldBackupNewDeployment = func() {
+			By("Creating Deployment " + deployment.Name)
+			_, err = f.CreateDeployment(deployment)
+			Expect(err).NotTo(HaveOccurred())
+			err = util.WaitUntilDeploymentReady(f.KubeClient, deployment.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Creating sample data inside workload")
+			err = f.CreateSampleDataInsideWorkload(deployment.ObjectMeta, apis.KindDeployment)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Reading sample data from /source/data mountPath inside workload")
+			sampleData, err = f.ReadSampleDataFromFromWorkload(deployment.ObjectMeta, apis.KindDeployment)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(sampleData).To(Not(BeEmpty()))
+
+			By("Creating storage Secret " + cred.Name)
+			err = f.CreateSecret(cred)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Creating new repository")
+			err = f.CreateRepository(repo)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Creating BackupConfiguration" + backupCfg.Name)
+			err = f.CreateBackupConfiguration(backupCfg)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Waiting for sidecar")
+			f.EventuallyDeployment(deployment.ObjectMeta).Should(matcher.HaveSidecar(util.StashContainer))
+
+			By("Waiting for BackupSession")
+			f.EventuallyBackupSessionCreated(backupCfg.ObjectMeta).Should(BeTrue())
+			bs, err := f.GetBackupSession(backupCfg.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Check for succeeded BackupSession")
+			f.EventuallyBackupSessionPhase(bs.ObjectMeta).Should(Equal(v1beta1.BackupSessionSucceeded))
+
+			By("Check for repository status updated")
+			f.EventuallyRepository(&deployment).Should(WithTransform(f.BackupCountInRepositoriesStatus, BeNumerically(">=", 1)))
+
+			By("Delete BackupConfiguration")
+			err = f.DeleteBackupConfiguration(backupCfg)
+			err = framework.WaitUntilBackupConfigurationDeleted(f.StashClient,backupCfg.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Waiting to remove sidecar")
+			f.EventuallyDeployment(deployment.ObjectMeta).ShouldNot(matcher.HaveSidecar(util.StashContainer))
+			err = util.WaitUntilDeploymentReady(f.KubeClient, deployment.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+
+		}
+	)
+	Context("Backup && Restore for Deployment", func() {
+		BeforeEach(func() {
+			pvc = f.GetPersistentVolumeClaim()
+			err = f.CreatePersistentVolumeClaim(pvc)
+			Expect(err).NotTo(HaveOccurred())
+			deployment = f.Deployment(pvc.Name)
+
+			targetref = v1beta1.TargetRef{
+				APIVersion: "apps/v1",
+				Kind:       apis.KindDeployment,
+				Name:       deployment.Name,
+			}
+		})
+		AfterEach(func() {
+			err := f.DeleteDeployment(deployment.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+			f.DeleteRepositories(f.DeploymentRepos(&deployment))
+			err =framework.WaitUntilDeploymentDeleted(f.KubeClient,deployment.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+
+		})
+		It("General Backup new Deployment", func() {
+			shouldBackupNewDeployment()
```

Comment:
Add a `By()` or some comment explaining what are you running here. Also add a new line after this line to make it noticeable.